api/v1/report.proto

```diff
@@ -18,131 +18,18 @@ package istio.mixer.v1;
 
 option go_package="istio.io/mixer/api/v1;mixerpb";
 
-import "google/protobuf/timestamp.proto";
-import "google/protobuf/struct.proto";
-import "google/protobuf/any.proto";
+import "attributes.proto";
 import "google/rpc/status.proto";
 
 // Used to report telemetry after performing an action.
 message ReportRequest {
   // Uniquely identifies this request, used to match to responses
   int64 request_id = 1;
 
-  // The set of discovered facts associated with this request. This value
```

Comment:
Why would we need to do that?

Can you explain a use case where you would need to define a separate method for non-proxy callers?

The direction we're going with the api and config is that the caller should not need a lot of custom logic, and instead mixer should do the heavy lifting, so I'm curious why a more complex API is necessary.